core/vision/src/main/java/com/arcrobotics/ftclib/vision/UGBasicHighGoalPipeline.java

```diff
@@ -73,43 +89,36 @@ public Mat processFrame(Mat input) {
 
         Imgproc.findContours(blueThreshold, blueContours, new Mat(), Imgproc.RETR_TREE, Imgproc.CHAIN_APPROX_SIMPLE);
         Imgproc.findContours(redThreshold, redContours, new Mat(), Imgproc.RETR_TREE, Imgproc.CHAIN_APPROX_SIMPLE);
+        blueContours = blueContours.stream().filter(i -> filterContours(i) && (((double) Imgproc.boundingRect(i).width / Imgproc.boundingRect(i).height) > 1) &&(((double) Imgproc.boundingRect(i).width / Imgproc.boundingRect(i).height) < 2)).collect(Collectors.toList());
+        redContours = redContours.stream().filter(i -> filterContours(i) && (((double) Imgproc.boundingRect(i).width / Imgproc.boundingRect(i).height) > 1) &&(((double) Imgproc.boundingRect(i).width / Imgproc.boundingRect(i).height) < 2)).collect(Collectors.toList());
 
+        Imgproc.drawContours(input, redContours, -1, new Scalar(255, 255, 0));
         if (blueContours.size() != 0) {
             biggestBlueContour = Collections.max(blueContours, new Comparator<MatOfPoint>() {
                 @Override
                 public int compare(MatOfPoint t0, MatOfPoint t1) {
-                    return Double.compare(Imgproc.contourArea(t0), Imgproc.contourArea(t1));
+                    return Double.compare(Imgproc.boundingRect(t0).width, Imgproc.boundingRect(t1).width);
```

Comment:
Can you elaborate on this change? What does it do differently? I assume this is for avoiding the issue with it detecting the 4 rings as a high goal.

Reply:
Comparing width instead of area because wobble goals that are close to the camera tend to have a large area